squidb/src/com/yahoo/squidb/sql/Function.java

```diff
@@ -246,6 +246,9 @@ protected String separator() {
      * Begins a CASE statement, populating it with the first WHEN ... THEN branch
      */
     public static CaseBuilder caseWhen(Criterion when, Object then) {
+        if (when == null) {
+            throw new IllegalArgumentException("Can't construct a CASE WHEN statement with a null criterion");
```

Comment:
We did this and then realized we were wrong. "when null" is sometimes valid, e.g. "case x when null then 0 ..." So we only disallow null for criterions in the "case when <criterion>" case. If you really want "case when null", you can use Criterion.literal(null).